homeassistant/components/plex/sensor.py

```diff
@@ -99,11 +104,25 @@ def device_state_attributes(self):
     def update(self):
         """Update method for Plex sensor."""
         _LOGGER.debug("Refreshing sensor [%s]", self.unique_id)
+
+        @callback
+        def update_plex(_):
+            async_dispatcher_send(
+                self.hass,
+                PLEX_UPDATE_PLATFORMS_SIGNAL.format(self._server.machine_identifier),
+            )
+
         now_playing = []
         for sess in self.sessions:
             if sess.TYPE == "photo":
                 _LOGGER.debug("Photo session detected, skipping: %s", sess)
                 continue
+            if not sess.usernames:
+                _LOGGER.debug(
+                    "Session temporarily incomplete, will try again: %s", sess
+                )
+                async_call_later(self.hass, 5, update_plex)
```

Comment:
Does this need a debouncer ? Now it could end up scheduling a lot of these calls ?

Reply:
This signal will go through the existing debouncer.

I guess there's a risk there's an environment that _never_ returns anything in `usernames`, but every log I've seen so far this shows up as a transient issue. Worst case (which I don't expect to happen) it would update every 5 seconds.

Reply:
If 2 sessions are incomplete, we will now schedule 2 call_later. Let's make sure that's only 1.

Reply:
You are calling an async function inside a sync context.

Reply:
Just returning now instead of continuing through the loop if any session is incomplete.